### PR TITLE
DSD-865: Reservoir branding

### PR DESCRIPTION
**Pull request by bigfishdesign13**
Fixes JIRA ticket [DSD-865](https://jira.nypl.org/browse/DSD-865)

## This PR does the following:

- Adds `Reservoir` branding to Storybook.
- Updates all references to "design system" in the Storybook documentation to include "Reservoir" in the name of the design system.
- Updates language on the `Welcome` page.

## How has this been tested?

- n/a

### Checklist:

- [x] I have updated the Storybook documentation accordingly.
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr922-bin804gtjifjbmgnazrd3nuseuibkkh5.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624309c29413e8006b2ca6a9

File: .storybook/CustomTheme.js [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import { create } from "@storybook/theming";
-import logo from "./public/nypl-logo.svg";
+import logo from "./public/nypl-reservoir-design-system-logo.svg";
 
 export default create({
   base: "light",
-  brandTitle: "NYPL Design System React Library",
-  brandUrl: "https://nypl.org",
+  brandTitle: "Reservoir Design System React Library",
```
Comment (EdwinGuzman) [inline .storybook/CustomTheme.js]: We're not adding "NYPL" before Reservoir?

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,12 +12,15 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
 
 - Adds a `contentId` prop to the `TemplateAppContainer` component and adds an `id` prop to the `TemplateContent` component. The default value of these `id`s are both set to "mainContent" and it will render as an attribute on the `main` HTML element. This is used as the target for the skip navigation link in consuming applications.
 - Adds an `Accessibility Guide` section to Storybook with a "Skip Navigation" page.
+- Adds `Reservoir` branding to Storybook.
```
Comment (EdwinGuzman) [inline CHANGELOG.md]: The new additions should be at the end of the list to keep the chronological order of when they were added into `development`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,12 +12,15 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
...
 - Adds the `isFullWidth` prop to the `CheckboxGroup` and `RadioGroup` components. This sets the wrapper element to be full width for labels that need to span its container.
 
 ### Changes
 
 - Updates Storybook's sidebar categories and documentation.
 - Updates the `Image`'s caption font size to "12px" (`text.tag`).
+- Updates all references to "design system" in the Storybook documentation to include "Reservoir" in the name of the design system.
+- Updates language on the `Welcome` page.
```
Comment (EdwinGuzman) [inline CHANGELOG.md]: Same comment for these.

File: src/components/StyleGuide/DesignTokens.stories.mdx [hunk 4]
```diff
@@ -18,23 +18,39 @@ import { getCategory } from "../../utils/componentCategories";
...
+_"With design tokens, you can capture low-level values and then use them to
+create the styles for your product or app. You can maintain a scalable and
+consistent visual system for UI development."_ - Jina Bolton, Lead Designer
+for the Salesforce Design System
 
-Design tokens are a single source of truth when it comes to styles. They allow an organization to more easily build multiple applications – web and/or native – that feel and look the same without slowing down the development team.
+Design tokens are a single source of truth when it comes to styles. They allow
+an organization to more easily build multiple applications - web and/or native
+
+- that feel and look the same without slowing down the development team.
```
Comment (EdwinGuzman) [inline src/components/StyleGuide/DesignTokens.stories.mdx]: The hyphen adds this line as a bullet point.

File: src/components/StyleGuide/Forms.stories.mdx [hunk 5]
```diff
@@ -6,7 +6,7 @@ import List from "../List/List";
 import { ListTypes } from "../List/ListTypes";
 import { getCategory } from "../../utils/componentCategories";
 
-<Meta title={getCategory("Forms")} />
+<Meta title={getCategory("Forms")} decorators={[withDesign]} />
```
Comment (EdwinGuzman) [inline src/components/StyleGuide/Forms.stories.mdx]: This is only needed if the figma file will be displayed in the canvas addon section, which in this case will not be displayed.

File: src/docs/Welcome.stories.mdx [hunk 6]
```diff
@@ -1,53 +1,172 @@
...
+      <Text>
+        Reservoir is NYPL's open-source extensible React library for products
+        and experiences, with accessibility as its core. With an aim to provide
+        design and user experience consistency, it is the foundation for all
+        digital properties in the NYPL landscape, offering functional components
+        that implement the NYPL style standards.
+      </Text>
+    </Box>
+    <SimpleGrid>
+      <div
```
Comment (EdwinGuzman) [inline src/docs/Welcome.stories.mdx]: Maybe use `Box` instead of `div`?

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
Some minor changes and questions. The only place I think this update is missing is in the `package.json` file under the "description" key.

File: .storybook/CustomTheme.js (hunk 1 above)
Comment (bigfishdesign13) [reply .storybook/CustomTheme.js]: Yes, I will add `NYPL` here.

File: src/components/StyleGuide/DesignTokens.stories.mdx (hunk 4 above)
Comment (bigfishdesign13) [reply src/components/StyleGuide/DesignTokens.stories.mdx]: Ha!  Fixed.

File: src/components/StyleGuide/Forms.stories.mdx (hunk 5 above)
Comment (bigfishdesign13) [reply src/components/StyleGuide/Forms.stories.mdx]: Ah, good catch.  I added that because I was trying to figure out why the fonts are not rendering properly on some of the Style Guide pages, `Form` being on of them.  I am still working to figure that out, but I will remove this.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr922-bin804gtjifjbmgnazrd3nuseuibkkh5.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624309c29413e8006b2ca6a9

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr922-bin804gtjifjbmgnazrd3nuseuibkkh5.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624309c29413e8006b2ca6a9

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr922-bin804gtjifjbmgnazrd3nuseuibkkh5.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624309c29413e8006b2ca6a9

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr922-bin804gtjifjbmgnazrd3nuseuibkkh5.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624309c29413e8006b2ca6a9

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr922-bin804gtjifjbmgnazrd3nuseuibkkh5.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624309c29413e8006b2ca6a9

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr922-bin804gtjifjbmgnazrd3nuseuibkkh5.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624309c29413e8006b2ca6a9

File: package.json [hunk 7]
```diff
@@ -32,7 +32,7 @@
     "size": "size-limit",
     "analyze": "size-limit --why",
     "generate-sass-resources": "gulp",
-    "storybook": "start-storybook -p 6006",
+    "storybook": "start-storybook -p 6006 -s ./.storybook/public",
```
Comment (EdwinGuzman) [inline package.json:35]: Why was this added?

**Review by EdwinGuzman — COMMENTED** [review]
Everything looks good but I'm wondering about the storybook command.

File: package.json (hunk 7 above)
Comment (bigfishdesign13) [reply package.json:35]: @EdwinGuzman In order to reference the favicon.

Comment (bigfishdesign13) [reply package.json:35]: In here:
`.storybook/manager-head.html`

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr922-bin804gtjifjbmgnazrd3nuseuibkkh5.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624309c29413e8006b2ca6a9

**Review by EdwinGuzman — APPROVED** [review]
